Realm/RLMQueryUtil.mm

```diff
@@ -980,6 +989,34 @@ void update_query_with_column_expression(RLMSchema *schema, RLMObjectSchema *des
                             std::move(left), std::move(right));
 }
 
+NSExpression *simplify_self_value_for_key_path_function_expression(NSExpression *expression) {
+    // -[NSPredicate predicateWithSubtitutionVariables:] results in function expressions of the form [SELF valueForKeyPath:]
+    // that update_query_with_predicate cannot handle. Replace such expressions with equivalent NSKeyPathExpressionType expressions.
+    if (expression.expressionType == NSFunctionExpressionType
+        && expression.operand.expressionType == NSEvaluatedObjectExpressionType
+        && [expression.function isEqualToString:@"valueForKeyPath:"]) {
+        if (NSString *keyPath = expression.arguments.firstObject.keyPath) {
```

Comment:
I find this indentation hard to read, as it's hard to scan where the conditional ends and the body of the block begins. I'd find it easier to read if the continuation of the condition was indented further, or perhaps even extracted to a separate function entirely (although in this case that'd render this function a bit pointless).